src/awkward/_connect/numba/arrayview_cuda.py

```diff
@@ -18,10 +18,16 @@ def prepare_args(self, ty, val, stream, retr):
                 # Use uint64 for pos, start, stop, the array pointers values, and the pylookup value
                 tys = numba.types.UniTuple(numba.types.uint64, 5)
 
-                start = val._numbaview.start
-                stop = val._numbaview.stop
-                pos = val._numbaview.pos
-                arrayptrs = val._numbaview.lookup.arrayptrs.data.ptr
+                if val._numbaview is None:
```

Comment:
@jpivarski - when a type is given as a parameter to a numba-cuda kernel an array view is not generated. It is generated first time when an array `numba_type` attribute is accessed.

Reply:
ok, I think we need to check the type here!

Reply:
The assumption was that Numba would need to know the `numba_type` in order to start compiling, so accessing the `numba_type` property makes the necessary ArrayView.

If `prepare_args` is a code path that doesn't always invoke `numba_type`, then manually invoking it like this sounds like a good solution.